sharding/collator.go

```diff
@@ -79,35 +79,35 @@ func checkShardsForProposal(c collatorClient, head *types.Header) error {
 		// If output is non-empty and the addr == coinbase
 		if addr == account.Address {
 			log.Info(fmt.Sprintf("Selected as collator on shard: %d", s))
-			err := proposeCollation(s)
+			err := submitCollation(s)
 			if err != nil {
-				return fmt.Errorf("could not propose collation. %v", err)
+				return fmt.Errorf("could not add collation. %v", err)
 			}
 		}
 	}
 
 	return nil
 }
 
-// isAccountInValidatorSet checks if the client is in the validator pool because
+// isAccountInCollatorPool checks if the client is in the collator pool because
 // we can't guarantee our tx for deposit will be in the next block header we receive.
-// The function calls IsValidatorDeposited from the VMC and returns true if
-// the client is in the validator pool
-func isAccountInValidatorSet(c collatorClient) (bool, error) {
+// The function calls IsCollatorDeposited from the SMC and returns true if
+// the client is in the collator pool
+func isAccountInCollatorPool(c collator) (bool, error) {
 	account := c.Account()
 
-	// Checks if our deposit has gone through according to the VMC
-	b, err := c.VMCCaller().IsValidatorDeposited(&bind.CallOpts{}, account.Address)
+	// Checks if our deposit has gone through according to the SMC
+	b, err := c.SMCCaller().IsCollatorDeposited(&bind.CallOpts{}, account.Address)
```

Comment:
Just return the result of this method call. No need for the following if else statement